### PR TITLE
[WIP] Attempt at improving MDN annotations

### PR DESCRIPTION
**Pull request by domenic**
@sideshowbarker this is an attempt at #94 with less duplicate code. However, it doesn't quite work. Let's discuss in #94.

### CODE REVIEW

**Comment by sideshowbarker** [comment]
When running html-build with a wattsi built from this branch, I get an Access violation at runtime.

**Comment by sideshowbarker** [comment]
OK after looking back at this, I agree it’s an improvement code-wise; but after testing, I find that it doesn’t seem to have fixed the problem that #94 was intended to fix — namely, that MDN annos for headings can end up orphaned in the previous (split) output file.

For example, take a look at the `multipage/media.html` output file which gets generated by the code in this branch. I think at the bottom of the page you’ll see an MDN anno that corresponds to the **The map element** heading — which is actually in the next file, `multipage/image-maps.html`.

Lemme see if I can look back now at my code on the #94 branch and try to take whatever specific part there was intended to fix that orphaned-anno-for-heading-in-next-outpuf-file problem, and port it to this branch…

**Comment by sideshowbarker** [comment]
Rebased + pushed an additional commit to the branch for causing the MDN annotations to be collapsed by default. I checked to make sure the rebase didn’t regress the behavior in the branch here, but if you can doublecheck and confirm, then I think we can go ahead and merge this.

**Comment by domenic** [comment]
Thanks so much for running with this!

Unfortunately right now when compiling this branch and running html-build I am seeing the output having _no_ MDN annotations :(. I'm not sure what went wrong...

**Comment by sideshowbarker** [comment]
> Unfortunately right now when compiling this branch and running html-build I am seeing the output having _no_ MDN annotations :(. I'm not sure what went wrong...

oh weird dunno how I managed to break it — or why it’d be working in my environment by not for you. I’ll investigate

**Comment by sideshowbarker** [comment]
> Unfortunately right now when compiling this branch and running html-build I am seeing the output having _no_ MDN annotations :(. I'm not sure what went wrong...

I can’t reproduce that. I just now created a fresh clone of the wattsi repo, checked out this branch, built wattsi, and ran html-build with it — and in the output I get MDN annotations as expected.

**Comment by domenic** [comment]
Hmm, OK. html-build is master, right? Let me try again, with fresh repos I guess...

**Comment by sideshowbarker** [comment]
> Hmm, OK. html-build is master, right?

Yep

**Comment by domenic** [comment]
Hmm, it works! I may have been misreading the diff. This knocks off 820 KiB!

It needs your approval to merge, so I'll leave it up to you :). Just remember to squash and write a nice commit message, probably by copying most of the OP from https://github.com/whatwg/wattsi/pull/94.

**Comment by sideshowbarker** [comment]
Cheers — thanks, will do
